features/frameworks/greentea-client/source/greentea_metrics.cpp

```diff
@@ -35,10 +35,13 @@ typedef struct {
 
 // Mutex to protect "buf"
 SingletonPtr<Mutex> mutex;
+#if defined(MBED_STACK_STATS_ENABLED) && MBED_STACK_STATS_ENABLED
 static char buf[128];
 static SingletonPtr<CircularBuffer<thread_info_t, THREAD_BUF_COUNT> > queue;
+#endif
 
 static void send_heap_info(void);
+#if defined(MBED_STACK_STATS_ENABLED) && MBED_STACK_STATS_ENABLED
```

Comment:
I've used the same construct as the one which was already there: see [here](https://github.com/ARMmbed/mbed-os/blob/master/features/frameworks/greentea-client/source/greentea_metrics.cpp#L61) and [there](https://github.com/ARMmbed/mbed-os/blob/f0c00bfd2101a9c56d5f44cca189d394e8db8933/rtos/rtx/TARGET_CORTEX_M/RTX_Conf_CM.c#L104).

I believe the intent was to check that `MBED_STACK_STATS_ENABLED` is defined before testing its value.